tvshows/src/components/shared/AuthForm/AuthForm.tsx

```diff
@@ -0,0 +1,246 @@
+"use client";
+import { IRegisterFormData, ISignInFormData } from "@/typings/authForms.types";
+import {
+  Alert,
+  Box,
+  Button,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Link,
+  Text,
+  chakra,
+} from "@chakra-ui/react";
+import { useState } from "react";
+import { useForm, Path } from "react-hook-form";
+import validator from "validator";
+
+interface IAuthFormProps<T> {
+  title: string;
+  description: string;
+  submitButtonText: string;
+  redirectButtonText: string;
+  linkText: string;
+  linkHref: string;
+  onSubmit: (data: T) => void;
+  successMessage?: string;
+  submitError?: string;
+  confirmPassword: boolean; // also used to check if user is currently on registration form
+  submitted: boolean;
+}
+
+export const AuthForm = <T extends IRegisterFormData | ISignInFormData>({
```

Comment:
I love the "pro" approach to have the AuthForm component that handles two different cases. Unfortunately ,this causes some complications for you, one of them is that useForm struggles with generics: https://github.com/react-hook-form/react-hook-form/issues/6726. 
Let's discuss on our sync.

Reply:
I realized it causes complications, it was really tiring trying to get everything to work. I wanted to do this kind of approach because I realized the two forms are quite similar.